src/context-providers/builder/hook.ts

```diff
@@ -58,6 +58,39 @@ export const useBuilder = () => {
         [state.orderedIdentifiers]
     );
 
+    const duplicateElement = useCallback(
+        (element: TElement) => {
+            const existingIdentifiers = state.orderedIdentifiers.map(
+                (elementId) => elementId.internalId
+            );
+            const newElement = ElementObjectGenerator.duplicate(
+                element,
+                existingIdentifiers
+            );
+            const newOrderedIdentifiers = [
+                ...state.orderedIdentifiers,
+                { internalId: newElement.internalId },
+            ];
+
+            dispatch({
+                type: "duplicate-element",
+                payload: {
+                    element: newElement as TElement,
+                    orderedIdentifiers: newOrderedIdentifiers,
+                },
+            });
```

Comment:
`duplicate-element` action appears redundant because it is not doing anything different from `add-element` so you can use `add-element` instead